hydrus/data/crud.py

```diff
@@ -570,12 +564,16 @@ def get_collection(API_NAME: str,
         if apply_filter(instance_.id, session, search_props=search_props) is True:
             filtered_instances.append(instance_)
     result_length = len(filtered_instances)
-
-    # To paginate, calculate offset and page_limit values for pagination of search results
-    page_limit, offset = calculate_page_limit_and_offset(paginate=paginate, page_size=page_size, page=page,
-                                                         result_length=result_length)
-
-    for i in range(offset, offset+page_limit):
+    try:
+        # To paginate, calculate offset and page_limit values for pagination of search results
+        page, page_size, offset = pre_process_pagination_parameters(search_params=search_params, paginate=paginate,
+                                                                 page_size=page_size, result_length=result_length)

```

Comment:
This line is 116 characters long, isn't the PEP8 test supposed to fail with this?